File: frontend/src/app/shared/services/paciente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Paciente } from '../models/paciente.model';

@Injectable({
  providedIn: 'root'
})
export class PacienteService {
  private apiUrl = 'http://localhost:8080/api/pacientes';

  constructor(private http: HttpClient) {}

  listarTodos(): Observable<Paciente[]> {
    return this.http.get<Paciente[]>(this.apiUrl);
  }

  listarInactivos(): Observable<Paciente[]> {
    return this.http.get<Paciente[]>(`${this.apiUrl}/inactivos`);
  }

  obtenerPorId(id: number): Observable<Paciente> {
    return this.http.get<Paciente>(`${this.apiUrl}/${id}`);
  }

  registrar(paciente: Paciente): Observable<Paciente> {
    return this.http.post<Paciente>(this.apiUrl, paciente);
  }

  actualizar(id: number, paciente: Paciente): Observable<Paciente> {
    return this.http.put<Paciente>(`${this.apiUrl}/${id}`, paciente);
  }

  eliminarLogico(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  reactivar(id: number): Observable<void> {
    return this.http.patch<void>(`${this.apiUrl}/${id}/reactivar`, {});
  }
}

File: src/main/java/com/example/GestionClinica/service/PacienteService.java
package com.example.GestionClinica.service;

import com.example.GestionClinica.dto.PacienteDTO;
import java.util.List;

public interface PacienteService {
    List<PacienteDTO> listarTodos();
    List<PacienteDTO> listarInactivos();
    PacienteDTO obtenerPorId(Long id);
    PacienteDTO registrar(PacienteDTO dto);
    PacienteDTO actualizar(Long id, PacienteDTO dto);
    void eliminarLogico(Long id);
    void reactivarPaciente(Long id);
}